src/evolution_graphics.c

```diff
@@ -0,0 +1,640 @@
+}
+
+static void EvoTask_CreatePreEvoSparklesSet2(u8 taskId)
+{
+    u8 i;
+    if (gTasks[taskId].data[15] < 96)
+    {
+        if (gTasks[taskId].data[15] < 6)
+        {
+            for (i = 0; i < 9; i++)
+                CreatePreEvoSparkleSet2(16 * i);
+        }
+        gTasks[taskId].data[15]++;
+    }
+    else
+    {
+        gTasks[taskId].func = EvoTask_PreEvoSparkleSet2Teardown;
+    }
+}
+
+static void EvoTask_PreEvoSparkleSet2Teardown(u8 taskId)
+{
+    DestroyTask(taskId);
+}
+
+u8 LaunchTask_PostEvoSparklesSet1(void)
+{
+    u8 taskId = CreateTask(EvoTask_PostEvoSparklesSet1Init, 0);
+    return taskId;
+}
+
+static void EvoTask_PostEvoSparklesSet1Init(u8 taskId)
+{
+    SetEvoSparklesMatrices();
+    gTasks[taskId].data[15] = 0;
+    gTasks[taskId].func = EvoTask_CreatePostEvoSparklesSet1;
+    PlaySE(SE_REAPOKE);
+}
+
+static void EvoTask_CreatePostEvoSparklesSet1(u8 taskId)
+{
+    u8 i;
+    if (gTasks[taskId].data[15] < 48)
+    {
+        if (gTasks[taskId].data[15] == 0)
+        {
+            for (i = 0; i < 16; i++)
+            {
+                CreatePostEvoSparkleSet1(i * 16, 4);
+            }
+        }
+        if (gTasks[taskId].data[15] == 32)
+        {
+            for (i = 0; i < 16; i++)
+            {
+                CreatePostEvoSparkleSet1(i * 16, 8);
+            }
+        }
+        gTasks[taskId].data[15]++;
+    }
+    else
+    {
+        gTasks[taskId].func = EvoTask_PostEvoSparklesSet1Teardown;
+    }
+}
+
+static void EvoTask_PostEvoSparklesSet1Teardown(u8 taskId)
+{
+    DestroyTask(taskId);
+}
+
+u8 LaunchTask_PostEvoSparklesSet2AndFlash(u16 a0)
+{
+    u8 taskId = CreateTask(EvoTask_PostEvoSparklesSet2Init, 0);
+    gTasks[taskId].data[2] = a0;
+    return taskId;
+}
+
+static void EvoTask_PostEvoSparklesSet2Init(u8 taskId)
+{
+    SetEvoSparklesMatrices();
+    gTasks[taskId].data[15] = 0;
+    sub_80D0474();
+    CpuCopy16(&gPlttBufferFaded[32], &gPlttBufferUnfaded[32], 96);
+    BeginNormalPaletteFade(0xFFF90F1C, 0, 0, 16, RGB_WHITE);
+    gTasks[taskId].func = EvoTask_CreatePostEvoSparklesSet2;
+    PlaySE(SE_W080);
+}
+
+static void EvoTask_CreatePostEvoSparklesSet2(u8 taskId)
+{
+    u8 i;
+    if (gTasks[taskId].data[15] < 128)
+    {
+        switch (gTasks[taskId].data[15])
+        {
+        default:
+            if (gTasks[taskId].data[15] < 50)
+                CreatePostEvoSparkleSet2(Random() & 7);
+            break;
+        case 0:
+            for (i = 0; i < 8; i++)
+                CreatePostEvoSparkleSet2(i);
+            break;
+        case 32:
+            BeginNormalPaletteFade(0xFFFF0F1C, 16, 16, 0, RGB_WHITE);
+            break;
+        }
+        gTasks[taskId].data[15]++;
+    }
+    else
+    {
+        gTasks[taskId].func = EvoTask_PostEvoSparklesSet2Teardown;
+    }
+}
+
+static void EvoTask_PostEvoSparklesSet2Teardown(u8 taskId)
+{
+    if (!gPaletteFade.active)
+        DestroyTask(taskId);
+}
+
+u8 LaunchTask_PostEvoSparklesSet2Alt(u16 a0)
```

Comment:
in the header there's a function called LaunchTask_PostEvoSparklesSet2AndFlash_Trade, is it the same function?